File: src/luogu/p1012/Main.cpp
// https://www.luogu.com.cn/problem/P1012
#include <bits/stdc++.h>
using namespace std;
int main() {
  int n;
  cin >> n;
  int ls[n];
  for (int i = 0; i < n; i++) {
    cin >> ls[i];
  }
  sort(ls, ls + n, [](int a, int b) {
    string sa = to_string(a) + to_string(b);
    string sb = to_string(b) + to_string(b);
    return sa > sb;
  });
  for (auto&& i : ls) {
    cout << i;
  }
  return 0;
}

File: src/luogu/p1012/Main.java
// https://www.luogu.com.cn/problem/P1012
package luogu.p1012;

import java.util.Arrays;
import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    Integer[] ls = new Integer[n];
    for (int i = 0; i < n; i++) {
      ls[i] = sc.nextInt();
    }
    Arrays.sort(ls, (Integer a, Integer b) -> {
      String sa = String.valueOf(a) + String.valueOf(b);
      String sb = String.valueOf(b) + String.valueOf(a);
      return -sa.compareTo(sb);
    });
    for (int integer : ls) {
      System.out.print(integer);
    }
    sc.close();
  }
}
